### ISSUE TITLE
🌸 Cherry-pick request for #30783 into #30741, #30770 (Approved)

### ISSUE THREAD

**Issue by Enriqe** [ampproject/amphtml#30805]
# Cherry-pick request

|       Issue       |       PR       | Beta / Experimental? |   Stable?    |     LTS?     | [Release issue](https://github.com/ampproject/amphtml/labels/Type%3A%20Release) |
| :---------------: | :------------: | :------------------: | :----------: | :----------: | ------------------------------------------------------------------------------- |
| #30783 | #30804 |     **YES**     | **YES** | **NO** | #30741 #30770                                                             |

## Why does this issue meet the [cherry-pick criteria](https://github.com/ampproject/amphtml/blob/master/contributing/contributing-code.md#Cherry-picks)?

Stories are not being rendered in all `amp-story-player`s. 

## List the steps to manually verify the changes in this cherry-pick

1. Load a local amp-story-player in a document with the `DOCTYPE` tag.
2. Fix the CSS like described on the PR.
3. Issue should be fixed

# Mini-postmortem

## Summary

The issue was caused by the introduction of an intermediate element in #30433.

Introducing a new element between the `<amp-story-player>` and the Shadow DOM was meant to fix the responsive behavior in the AMP version of the player. The problem was that I mistakenly added the styles for this intermediate element in our stylesheets for the shadow DOM, not for the light DOM. And because this element is not part of the shadow DOM, it wasn't getting any of the corresponding styles, including its `height`. By not having an explicit `height`, it was defaulting to `0`, causing the story to not be displayed.

We didn't catch this during manual / QA testing because we were using documents without the `<!DOCTYPE html>` tag, which switches from [quirks mode to standards mode](https://stackoverflow.com/questions/1966300/height-100-is-not-working-in-html-when-using-doctype). In standards mode, the `html` and `body` elements do not default to 100%, but in quirks mode they do. By testing in quirks mode, the intermediate element was actually getting 100% height, which was making it look like it worked properly.

On a final note, I'd like to point out that even after a few hours of the rollback being deployed, sometimes I would still get the release containing the faulty code. 

## Impact

- All `amp-story-player` in documents not using quirks mode were affected.

## Action Items

* Updating all tests and manual QA pages to include the <!DOCTYPE html> tag. 
* To help debugging easier/faster: Make sure local URL rewriting to the local build works for the cdn.ampproject.org/amp-story-player-v0.js (Right we have to use the relative path to the dist/ folder to get the local build). https://github.com/ampproject/amphtml/issues/30817

---

/cc @ampproject/release-on-duty @ampproject/wg-approvers @ampproject/cherry-pick-approvers

**Comment by cramforce** [comment]
Approved. However, I'm not sure it technically needs to go into stable since stable was rolled back.

**Comment by Enriqe** [comment]
Correct, it just needs to be CP into the rolled back release (2010122315000) and then we can proceed with a rollforward.

**Comment by caroqliu** [comment]
Cherry-picks are complete. @Enriqe please verify.

**Comment by Enriqe** [comment]
I verified the fix is in production by trying out several players in various URLs. Will close this issue once PM is complete.

Thanks @caroqliu for all your work here!
